core/ir/aarch64/instr_create_api.h

```diff
@@ -597,7 +600,8 @@
  * result of a comparison of its two source values if the named input condition
  * is true, or to an immediate value if the input condition is false.
  * \param dc      The void * dcontext used to allocate memory for the #instr_t.
- * \param cond    The comparison condition specified by #dr_pred_type_t, e.g. #DR_PRED_EQ.
+ * \param cond    A 4-bit immediate value created using #OPND_CREATE_COND() specifying
+ *                the condition used for the comparison
```

Comment:
This still holds a value that can be interpreted as of type `#dr_pred_type_t`, right?  We should keep that part of the doc and link to the pred type?  (Ditto 4x below.)

Reply:
Unfortunately not. `dr_pred_type_t` begins with `DR_PRED_NONE` so all the `DR_PRED_<cond>` members are off by one compared to the immediate.

Reply:
That is unfortunate...I think we should have `OPND_CREATE_COND` take in `dr_pred_type_t` and adjust it.

Reply:
I have replaced the macro with a function that does the adjustment.